### PR TITLE
Adding cropping option for SSIM

### LINKED ISSUE #1204: [Feature] Add padding equal to None option for SSIM to match the original MATLAB implementation
## 🚀 Feature
Add padding equal to None option for SSIM  to match the original MATLAB implementation

## Motivation
I am comparing `kornia` and `skimage` for computing SSIM. I find that there is always a slight difference between the two results. I read the code and found that the differences come from `kornia` using the reflect padding in the gaussian weighting function to keep the size of the image the same, while `skimage` doesn't use padding here. I also check [the MATLAB implementation](http://www.cns.nyu.edu/~lcv/ssim/ssim_index.m) of SSIM from the original authors, they also don't use padding. 

When I remove the reflect padding in `kornia`, the SSIM would return a more similar result with `skimage`.  So I think it would be better if we can add an option in the SSIM in `kornia` to be able to remove padding. 

The related colab link is as following: https://colab.research.google.com/drive/1JCDQn_OWbm52hyDadKiU1UpwB3glGY5f?usp=sharing

A lot of thanks for this wonderful project!

### PR DESCRIPTION
**Pull request by wendy-xiaozong**
#### Changes
Add cropping option for SSIM to crop out the "valid" convolution part to match the original MATLAB implementation.

Fixes #1204 

#### Type of change

- [X] 📚  Documentation Update
- [X] 🔬 New feature (non-breaking change which adds functionality)

#### Checklist
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings

### CODE REVIEW

File: kornia/metrics/ssim.py [hunk 1]
```diff
@@ -68,14 +83,29 @@ def ssim(
     mu1: torch.Tensor = filter2d(img1, kernel)
     mu2: torch.Tensor = filter2d(img2, kernel)
```
Comment (shijianjian) [inline kornia/metrics/ssim.py:84]:
```suggestion
    mu1: torch.Tensor = filter2d(img1, kernel, pad="valid")
    mu2: torch.Tensor = filter2d(img2, kernel, pad="valid")
```

I think applying here seem to be more reasonable to align with other libs.

File: kornia/losses/ssim.py [hunk 2]
```diff
@@ -75,6 +78,8 @@ class SSIMLoss(nn.Module):
          output: ``'none'`` | ``'mean'`` | ``'sum'``. ``'none'``: no reduction will be applied,
          ``'mean'``: the sum of the output will be divided by the number of elements
          in the output, ``'sum'``: the output will be summed.
+        cropping: Whether to crop out the "valid" convolution area to match
```
[the remark this answers is not in the document]
Comment (wendy-xiaozong) [reply kornia/losses/ssim.py]: Thanks for the comments! I have changed it. It would be clearer to name it in this way.

File: kornia/metrics/ssim.py [hunk 3]
```diff
@@ -1,11 +1,24 @@
+from typing import List
+
 import torch
 import torch.nn as nn
 
 from kornia.filters import filter2d, get_gaussian_kernel2d
+from kornia.filters.filter import _compute_padding
+
+
+def _crop(img: torch.Tensor, cropping_shape: List[int]) -> torch.Tensor:
```
[the remark this answers is not in the document]
Comment (wendy-xiaozong) [reply kornia/metrics/ssim.py:10]: In the corresponding issue, they think this is a simpler way to implement this. See: https://github.com/kornia/kornia/issues/1204#issuecomment-902143577

File: kornia/metrics/ssim.py (hunk 1 above)
Comment (wendy-xiaozong) [reply kornia/metrics/ssim.py:84]: Yes, it is more reasonable... Oh, I see. Now in filter2d, there is a padding option now, before there is not. I will change it!

Another thing I also think is worth changing is that the gaussian 2d filter can be reduced to two 1d convolutions (I can see there already is a `filter2d_separable` function). It can reduce complexity if we use that instead of this `filter2d` function. Do you recommend me change it in the SSIM function?

Comment (edgarriba) [reply kornia/metrics/ssim.py:84]: If the numerical precision is kept, why not

**Review by edgarriba — COMMENTED** [review]
I suggest to add test for both modalities

File: kornia/metrics/ssim.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (shijianjian) [reply kornia/metrics/ssim.py:10]: Okay. I prefer using the predefined functions than implement it in our end though. You may perform
```
torch.nn.functional.pad(img, (-cropping_shape[2], -cropping_shape[3], -cropping_shape[0], -cropping_shape[1]))
```

File: kornia/metrics/ssim.py (hunk 1 above)
Comment (shijianjian) [reply kornia/metrics/ssim.py:84]: > Yes, it is more reasonable... Oh, I see. Now in filter2d, there is a padding option now, before there is not. I will change it!
> 
> Another thing I also think is worth changing is that the gaussian 2d filter can be reduced to two 1d convolutions (I can see there already is a `filter2d_separable` function). It can reduce complexity if we use that instead of this `filter2d` function. Do you recommend me change it in the SSIM function?

Agree with edgar that if it accelerates without numerical changes. We can always upgrade.

**Comment by edgarriba** [comment]
@shijianjian please merge once green light passes
